controllers/capabilities/backend_controller.go

```diff
@@ -74,12 +81,46 @@ func (r *BackendReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if backend.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		// Attempt to remove backend only if backend.Status.ID is present
+		if backend.Status.ID != nil {
+			requeue, err := r.removeBackend(backend.Spec.ProviderAccountRef, *backend.Status.ID, backend.Namespace, backend.Spec.SystemName)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			if requeue {
+				return ctrl.Result{RequeueAfter: requeueTime}, nil
+			}
+		} else {
+			return ctrl.Result{}, fmt.Errorf("backend %s .status.ID is missing, cannot remove backend", backend.Spec.Name)
```

Comment:
This will start some endless loop. 

What about logging internally that the backend could not be removed because ID was missing?

Reply:
Good catch, thanks, logging internally seems like a way to go.